### PR TITLE
Exclude DIETClassifier from extractors in tests if no entities were trained

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
`DIETClassifier` only counts as extractor if `ENTITY_RECOGNITION` is set to `True`

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/test.py [hunk 1]
```diff
@@ -1022,12 +1023,17 @@ def get_entity_extractors(interpreter: Interpreter) -> Set[Text]:
...
-
     from rasa.nlu.extractors.extractor import EntityExtractor
 
-    extractors = {
-        c.name for c in interpreter.pipeline if isinstance(c, EntityExtractor)
-    }
+    extractors = set()
+    for c in interpreter.pipeline:
+        if isinstance(c, EntityExtractor):
+            if c.name == "DIETClassifier":
```
Comment (Ghostvv) [inline rasa/nlu/test.py:1031]: isn't it better?
```suggestion
            if isinstance(c, DIETClassifier):
```

Comment (Ghostvv) [reply rasa/nlu/test.py:1031]: shall we rather create `EntityExtractor.active` property? to make it DIET independent?

Comment (tabergma) [reply rasa/nlu/test.py]: Not sure if that is needed. Seems to be a bit too much as we just have DIET as a special case for now. Would reiterate on that once we have another edge case.
